src/components/buttons/default.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react'
+import styled, { css } from 'styled-components'
+import colors from '../../assets/colors'
+import * as fonts from '../../assets/fonts'
+
+interface ButtonProps extends React.HTMLProps<Button> {
+    state?: ButtonState
+    block?: boolean
+}
+
+export class ButtonState extends String {
+    static Default: ButtonState = "default"
+    static Loading: ButtonState = "loading"
+    static Success: ButtonState = "success"
+    static Failure: ButtonState = "failure"
+}
```

Comment:
Hah, this is some smart declaration 👌

However, I do wonder if having simply an enum of the various strings isn't better to grok?